Sources/HummingbirdLambdaTesting/FunctionURLLambda.swift

```diff
@@ -0,0 +1,89 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaEvents
+import Foundation
+import HTTPTypes
+import HummingbirdCore
+import NIOCore
+
+extension FunctionURLRequest: LambdaTestableEvent {
+    /// Construct APIGatewayV2 Event from uri, method, headers and body
+    public init(uri: String, method: HTTPRequest.Method, headers: HTTPFields, body: ByteBuffer?) throws {
+        let base64Body = body.map { "\"\(String(base64Encoding: $0.readableBytesView))\"" } ?? "null"
+        let url = URI(uri)
+        let queryValues: [String: [String]] = url.queryParameters.reduce([:]) { result, value in
+            var result = result
+            let key = String(value.key)
+            var values = result[key] ?? []
+            values.append(.init(value.value))
+            result[key] = values
+            return result
+        }
+        let queryValueStrings = try String(decoding: JSONEncoder().encode(queryValues.mapValues { $0.joined(separator: ",") }), as: UTF8.self)
+        let headerValues: [String: [String]] = headers.reduce(["host": ["127.0.0.1:8080"]]) { result, value in
+            var result = result
+            let key = String(value.name)
+            var values = result[key] ?? []
+            values.append(.init(value.value))
+            result[key] = values
+            return result
+        }
+        let headerValueStrings = try String(decoding: JSONEncoder().encode(headerValues.mapValues { $0.joined(separator: ",") }), as: UTF8.self)
+        let eventJson = """
+            {
+                "routeKey":"\(method) \(url.path)",
+                "version":"2.0",
+                "rawPath":"\(url.path)",
+                "stageVariables":null,
+                "requestContext":{
+                    "timeEpoch":1587750461466,
+                    "domainPrefix":"hello",
+                    "authorizer":{
+                        "jwt":{
+                            "scopes":[
```

Comment:
You should probably edit the authorizer to be one that a function URL can decode. ie an IAM one. I copied this from the AWS docs https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html
```jsonb
        "iam": {
                "accessKey": "AKIA...",
                "accountId": "111122223333",
                "callerId": "AIDA...",
                "cognitoIdentity": null,
                "principalOrgId": null,
                "userArn": "arn:aws:iam::111122223333:user/example-user",
                "userId": "AIDA..."
        }
```